arbos/programs/wasm.go

```diff
@@ -31,21 +28,20 @@ type usize = uintptr
 // opaque types
 type rustVec byte
```

Comment:
This isn't specific to this PR, but this should be renamed, as right now it sounds like it's the same as a RustVec but these are actually distinct types with different layouts. Actually I'd consider renaming RustVec instead because it can be confused with a rust Vec<u8>, even if this rustVec doesn't exist. Maybe call RustVec RustBytes instead or something?

Reply:
There's a mental separation in that `RustVec` is a cgo-only type. Nevertheless that makes sense, so I've gone ahead and renamed the cgo type to `RustBytes`